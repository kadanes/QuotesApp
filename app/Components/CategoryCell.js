import React, {Component} from 'react'
import { TouchableOpacity, View, Text, StyleSheet } from 'react-native'

export default class CategoryCell extends Component {
    render() {
        return (
            <TouchableOpacity 
                onPress={() => this.props.navigation.navigate('Quotes',{ id: this.props.id})}
            >
                <View style={styles.Container}>
                    <View style={styles.CategoryHeader}>
                        <Text style={styles.Category}> {this.props.Category.category} </Text>
                        <Text style={styles.QuotesCount}>{this.props.Category.quotes.length} </Text>
                    </View>
                    <View style={styles.AuthorBlock}>
                        {this.props.Category.quotes.map(item => <Text style={styles.Author}> {item.person}</Text>)}
                    </View>
                </View>
            </TouchableOpacity>
        )
    }
}


const styles = StyleSheet.create({
    Container: {
        margin: 10,
        padding: 10 ,
        backgroundColor: '#ff6347',
        borderRadius: 15,
        flex: 1
    },
    CategoryHeader: {
        flexDirection: 'row',
        flex: 1,
        justifyContent: 'space-between',
        
    },
    AuthorBlock: {
        flex: 1,
        flexDirection: 'column',
        marginTop: 10,
        marginLeft: 5
    },
    Category: {
        textTransform: 'capitalize',
        fontSize: 20,
        color: 'white',
        fontWeight: 'bold'
    },
    QuotesCount: {
        color: 'white',
        fontWeight: '600',
        fontSize: 12
    },
    Author: {
        color: '#E0E0E0',
        fontSize: 14
    }
})
