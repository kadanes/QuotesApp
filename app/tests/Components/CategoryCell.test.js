import React from 'react'
import CategoryCell from '../../Components/CategoryCell'
import { shallow } from 'enzyme'

import quotes from '../fixtures/quotes-fixture'

describe("testing CategoryCell component", () => {
    let wrapper,navigation

    beforeEach(() => {
    navigation = {
        navigate: jest.fn()
    }
    wrapper = shallow(<CategoryCell Category={quotes[0]} id={0} navigation={navigation} />);
    });

    it('category-cell renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should navigate with correct props',() => {
        wrapper.find('TouchableOpacity').first().props().onPress()
        expect(navigation.navigate).toHaveBeenLastCalledWith(
            'Quotes',
            {
                id: 0,
            }
        )
    })
})