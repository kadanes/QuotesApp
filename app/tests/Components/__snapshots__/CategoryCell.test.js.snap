// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CategoryCell component category-cell renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#ff6347",
        "borderRadius": 15,
        "flex": 1,
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "textTransform": "capitalize",
          }
        }
      >
         
        inspirational
         
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 12,
            "fontWeight": "600",
          }
        }
      >
        3
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginLeft": 5,
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#E0E0E0",
            "fontSize": 14,
          }
        }
      >
         
        Theodore Roosevelt
      </Text>
      <Text
        style={
          Object {
            "color": "#E0E0E0",
            "fontSize": 14,
          }
        }
      >
         
        Norman Vincent Peale
      </Text>
      <Text
        style={
          Object {
            "color": "#E0E0E0",
            "fontSize": 14,
          }
        }
      >
         
        Robert H. Schuller
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;
